content/blog/2025-07-21-valkey-on-k0rdent.md

```diff
@@ -0,0 +1,270 @@
+
+```bash
+kubectl label cluster docker-hosted-cp group=demo -n kcm-system
+```
+
+This label is needed because we will be using a MultiClusterService object that can reference multiple child clusters for service/application delivery. In our case, we will use our Docker-based cluster, still, we should keep in mind that we are not restricted as to which cluster we deliver new services, it can be a single child cluster or a group of them.
+
+Ok, let's do this!
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: k0rdent.mirantis.com/v1alpha1
+kind: MultiClusterService
+metadata:
+  name: valkey
+spec:
+  clusterSelector:
+    matchLabels:
+      group: demo
+  serviceSpec:
+    services:
+    - template: valkey-0-1-0
+      name: valkey
+      namespace: valkey-system
+      values: |
+        valkey:
+          spec:
+            tls: false # when enabled, needs CertManager (and some configs) inside child-cluster
+EOF
+```
+
+**Note:** In our case, 'values.valkey.spec' that are exposed inside the template are Valkey Operator Helm Chart values.
+
+## Verifying the Deployment
+
+Let's check the object status, we should see something similar to the example output:
+
+```bash
+kubectl get MultiClusterService -A
+```
+
+Expected output:
+```
+NAME     SERVICES   CLUSTERS   AGE
+valkey   1/1        1/1        23s
+```
+
+Now let's check how things look like inside the child cluster:
+
+```bash
+KUBECONFIG="docker-hosted-cp.kubeconfig" kubectl get pods -A
+```
+
+Expected output:
+```
+NAMESPACE              NAME                                                READY   STATUS    RESTARTS   AGE
+kube-system            coredns-5555f45c94-bf9mb                            1/1     Running   0          23m
+kube-system            konnectivity-agent-tfsr8                            1/1     Running   0          21m
+kube-system            kube-proxy-thx5h                                    1/1     Running   0          21m
+kube-system            kube-router-6b7s8                                   1/1     Running   0          21m
+kube-system            metrics-server-7778865875-s9hsz                     1/1     Running   0          23m
+local-path-storage     local-path-provisioner-74f9666bc9-5xqlf             1/1     Running   0          16m
+projectsveltos         sveltos-agent-manager-79df48c686-8l6dk              1/1     Running   0          23m
+valkey-system          valkey-0                                            1/1     Running   0          64s
+valkey-system          valkey-operator-controller-manager-6dc5d6bf57-rbt9x 1/1     Running   0          78s
+```
+
+See how application delivery is made very simple by k0rdent, pure magic!
+
+## Conclusion
+
+Feel free to play around with Valkey Operator by leveraging the MultiClusterService object together with additional Helm Chart values and when finished, cleaning up this environment is as simple as deleting the KinD cluster.
```

Comment:
'with Valkey Operator' -> 'with the Valkey Operator'

'MultiClusterService' -> '`MultiClusterService`'

'cleaning up this environment is as simple as deleting the KinD cluster' -> 'cleaning up the environment only requires deleting the KinD cluster'